ores/scoring_context.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import time
+from hashlib import sha1
+
+from revscoring import dependencies
+from revscoring.datasources import Datasource
+from revscoring.extractors import Extractor
+from revscoring.features import trim
+from revscoring.scorer_models import ScorerModel
+
+logger = logging.getLogger(__name__)
+
+
+class ScoringContext(dict):
+
+    def __init__(self, name, model_map, extractor):
+        """
+        :Parameters:
+            name : str
+               The  that this scorer is being used for
+            scorer_models : dict
+               A mapping between names and
+               :class:`revscoring.ScorerModel`
+               instances
+            extractor : :class:`revscoring.Extractor`
+               An extractor to use for gathering feature values
+        """
+        super().__init__()
+        self.name = str(name)
```

Comment:
Cheap runtime type checking like this is nice for catching issues.  I like str() and int() because they are free in the case that they encounter the right type.  

```
$ python
Python 3.5.1+ (default, Mar 30 2016, 22:46:26) 
[GCC 5.3.1 20160330] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> foo = "bar"
>>> id(foo)
139939565130224
>>> id(str(foo))
139939565130224
```